File: identhub/src/main/java/de/solarisbank/identhub/di/ActivityModuleContextFactory.java
package de.solarisbank.identhub.di;

import android.content.Context;

import de.solarisbank.identhub.di.internal.Factory;
import de.solarisbank.identhub.di.internal.Preconditions;

public final class ActivityModuleContextFactory implements Factory<Context> {

    private final ActivityModule activityModule;

    public ActivityModuleContextFactory(ActivityModule activityModule) {
        this.activityModule = activityModule;
    }

    public static ActivityModuleContextFactory create(ActivityModule activityModule) {
        return new ActivityModuleContextFactory(activityModule);
    }

    @Override
    public Context get() {
        return Preconditions.checkNotNull(
                activityModule.provideContext(),
                "Cannot return null from provider method"
        );
    }
}

File: core/src/main/java/de/solarisbank/sdk/core/di/ActivityModuleContextFactory.kt
package de.solarisbank.sdk.core.di

import android.content.Context
import de.solarisbank.sdk.core.di.internal.Factory
import de.solarisbank.sdk.core.di.internal.Preconditions

class ActivityModuleContextFactory(private val activityModule: ActivityModule) : Factory<Context> {
    override fun get(): Context {
        return Preconditions.checkNotNull(
                activityModule.provideContext(),
                "Cannot return null from provider method"
        )
    }

    companion object {
        @JvmStatic
        fun create(activityModule: ActivityModule): ActivityModuleContextFactory {
            return ActivityModuleContextFactory(activityModule)
        }
    }
}